### PR TITLE
Add document title for each page

### LINKED ISSUE #422: The title is always the same
I can't determine what page is what in my history, of if I have multiple tabs open, or any other situation when a opendota tab is opened in the background. The title is always "OpenDota". It gets set in the [original HTML](https://github.com/odota/ui/blob/master/index.html#L6) and I don't see it changed anywhere else.

### PR DESCRIPTION
**Pull request by masad-frost**
Fixes #422

### CODE REVIEW

File: src/components/App/App.jsx [hunk 1]
```diff
@@ -42,6 +44,11 @@ const App = ({ children, open, params, location, width }) => (
         ${location.pathname === '/' && styles.HomeBackground}`
       }
     >
+      <Helmet
+        htmlAttributes={{ lang: localization }}
+        defaultTitle="OpenDota - Dota 2 Statistics"
```
Comment (howardchung) [inline src/components/App/App.jsx]: These strings should go in the language file (english only)

File: src/components/Matches/index.jsx [hunk 2]
```diff
@@ -41,9 +42,12 @@ class RequestLayer extends React.Component {
     this.props.dispatchProMatches();
   }
   render() {
-    return (<Container>
-      <Table data={this.props.data} columns={matchesColumns} />
-    </Container>);
+    return (<div>
+      <Helmet title="Professional Matches" />
```
Comment (howardchung) [inline src/components/Matches/index.jsx]: put this in string file

File: src/components/Request/Request.jsx [hunk 3]
```diff
@@ -22,6 +23,7 @@ class Request extends React.Component {
       <CircularProgress value={progress} mode="indeterminate" />);
     return (
       <div style={{ textAlign: 'center' }}>
+        <Helmet title="Request a Parse" />
```
Comment (howardchung) [inline src/components/Request/Request.jsx]: string to string file

File: src/components/Search/Search.jsx [hunk 4]
```diff
@@ -13,7 +14,11 @@ class Search extends React.Component {
   }
   render() {
     const { data, pros, ...rest } = this.props;
-    return <SearchResult {...rest} players={data || []} pros={pros || []} />;
+    return (<div>
+      <Helmet title={`Search Results for ${this.props.location.query.q}`} />
```
Comment (howardchung) [inline src/components/Search/Search.jsx]: String to string file (use a template like %s and util.format to build the string here)

File: src/components/Status/Status.jsx [hunk 5]
```diff
@@ -26,6 +27,7 @@ class Status extends React.Component
   }
   render() {
     return (<div style={{ display: 'flex', flexDirection: 'row', flexWrap: 'wrap' }}>
+      <Helmet title="Status" />
```
Comment (howardchung) [inline src/components/Status/Status.jsx]: string to string file

File: src/components/Search/Search.jsx (hunk 4 above)
Comment (masad-frost) [inline src/components/Search/Search.jsx]: Anything against using template strings?

File: src/components/Search/Search.jsx [hunk 6]
```diff
@@ -13,7 +15,11 @@ class Search extends React.Component {
   }
   render() {
     const { data, pros, ...rest } = this.props;
-    return <SearchResult {...rest} players={data || []} pros={pros || []} />;
+    return (<div>
+      <Helmet title={`${strings.title_search} ${this.props.location.query.q}`} />
```
Comment (howardchung) [inline src/components/Search/Search.jsx]: The only potential problem with this is that the order may need to be rearranged for certain languages.  You might want something more generic like `${this.props.location.query.q} - ${strings.title_search}`
where title_search is "Search"

**Comment by masad-frost** [comment]
Done.

File: src/components/App/App.jsx [hunk 7]
```diff
@@ -42,6 +45,11 @@ const App = ({ children, open, params, location, width }) => (
         ${location.pathname === '/' && styles.HomeBackground}`
       }
     >
+      <Helmet
+        htmlAttributes={{ lang: localization }}
+        defaultTitle={strings.default_title}
```
Comment (blukai) [inline src/components/App/App.jsx]: Here `default_title`, in lang file - `title_default`

Comment (howardchung) [inline src/components/App/App.jsx]: I'll fix this post-merge.

Comment (masad-frost) [inline src/components/App/App.jsx]: Oh, sorry about that.

**Pull request merged**